### PR TITLE
[gosrc2cpg] - Unit test for array with handling for struct and pointer

### PR DESCRIPTION
**Pull request by ankit-privado**
Added unit test for array array along with handling for struct type and pointers. tests are failing for pointer.

### CODE REVIEW

File: joern-cli/frontends/gosrc2cpg/src/test/scala/io/joern/go2cpg/passes/ast/AstCreationForArraysTests.scala [hunk 1]
```diff
@@ -210,8 +210,200 @@ class AstCreationForArraysTests extends GoCodeToCpgSuite {
...
-//TODO: Add unit tests for pointer combinations for primitives as well as struct type
+  "be correct when array of struct within package same file" in {
+    val cpg = code("""
+        |package main
+        |
+        |type node struct {
+        |name string
+        |}
+        |
+        |func main() {
```
Comment (pandurangpatil) [inline joern-cli/frontends/gosrc2cpg/src/test/scala/io/joern/go2cpg/passes/ast/AstCreationForArraysTests.scala:222]: Following will be the valid code

```
func main() {
var a node = node{"value1"}
var b node = node{"value2"}

c := []node{a, b}
}
```

Comment (pandurangpatil) [reply joern-cli/frontends/gosrc2cpg/src/test/scala/io/joern/go2cpg/passes/ast/AstCreationForArraysTests.scala:222]: The current one has compilation errors

File: joern-cli/frontends/gosrc2cpg/src/test/scala/io/joern/go2cpg/passes/ast/AstCreationForArraysTests.scala [hunk 2]
```diff
@@ -210,8 +210,200 @@ class AstCreationForArraysTests extends GoCodeToCpgSuite {
...
+        |
+        |""".stripMargin,
+      "first.go"
+    )
+      .moreCode(
+        """
+          |package main
+          |
+          |func main() {
+          | a = node{"value1"}
```
Comment (pandurangpatil) [inline joern-cli/frontends/gosrc2cpg/src/test/scala/io/joern/go2cpg/passes/ast/AstCreationForArraysTests.scala:266]: same here

File: joern-cli/frontends/gosrc2cpg/src/test/scala/io/joern/go2cpg/passes/ast/AstCreationForArraysTests.scala [hunk 3]
```diff
@@ -210,8 +210,200 @@ class AstCreationForArraysTests extends GoCodeToCpgSuite {
...
+      .moreCode(
+        """
+          |package main
+          |
+          |import (
+          | "mypackage"
+          |)
+          |
+          |func main() {
+          | a = node{"value1"}
```
Comment (pandurangpatil) [inline joern-cli/frontends/gosrc2cpg/src/test/scala/io/joern/go2cpg/passes/ast/AstCreationForArraysTests.scala:316]: same here

**Review by pandurangpatil — CHANGES_REQUESTED** [review]
fix the compilation errors

**Review by DavidBakerEffendi — APPROVED** [review]

**Pull request merged**